Userland/Libraries/LibGfx/JPGLoader.cpp

```diff
@@ -1291,7 +1225,7 @@ ErrorOr<ImageFrameDescriptor> JPGImageDecoderPlugin::frame(size_t index)
         return Error::from_string_literal("JPGImageDecoderPlugin: Decoding failed");
 
     if (m_context->state < JPGLoadingContext::State::BitmapDecoded) {
-        if (!decode_jpg(*m_context)) {
+        if (decode_jpg(*m_context).is_error()) {
```

Comment:
Same note here about manual error checking / hiding the underlying error. This can also capture the result in a variable and then return result.release_error().